### DISCUSSION TITLE
S21 connector variant: PAP-05V-S pinout?

### DISCUSSION THREAD

**Discussion by basak** [revk/xESP32-Faikout#295] [General]
There seems to be a "newer" S21 connector that some models use which is JST PAP-05V-S, and the Daikin EKRS21 is the official way of converting it to the "old" S21 connector (JST EH-5) to connect to standard Daikin accessories. The current documentation doesn't really mention it, but some of the [working unit reports](https://github.com/revk/ESP32-Faikin/wiki/List-of-confirmed-working-air-con-units) mention this in passing, as do discussions #272 and #260.

I've connected a JST PAP-05V-S directly and am ready to connect my new Faikin to it, but I'm unsure of the pinout. Testing with a multimeter, it seems that ground is pin 1, and pin 4 against that is giving me 9.6V. Are the pins simply reversed? Or is there some more rearrangement? I'm counting pins based on the arrow to pin 1 on the PAP-05V-S connector.

I assume it'll be safe to connect the Faikin to my pins 1 and 4 for power, giving it 9.6V. Is there any known way I can safely identify the other pins please, without having any official S21 accessory? I do have a DSO available - what should I expect on the TX and RX pins when nothing is connected, if anything?

Or does anyone else with success with a PAP-05V-S but without the EKRS21 know the pinout they used please?

**Comment by basak** [comment]
> it seems that ground is pin 1

Actually pins 1, 2 and 3 all show about 9.6V below pin 4. From pin 5 all I seem to get is noise. So perhaps I have a bad connection on pin 5.

**Reply by revk** [reply]
I'm surprised if they have changed the pin assignments. But don't know for sure. Not on the Faikin pin 5 is GND.

**Comment by basak** [comment]
Turns out it indeed was a bad ground wire. Now fixed and communications seem to be working in both directions.

If it helps anyone else, the pinout is indeed the same for PAP-05V-S. TX and RX pins are both held high as I'd expect from a regular serial connection (which are usually active low), and so I see 5V against GND on pins 1, 2 and 3 as expected. Interestingly without the ground wire I wasn't something pretty low (like 0.1V or something) against pin 4 (12V) so I guess there's some isolation going on.
